cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -153,6 +153,11 @@ parse_numeric(const char* data, long start, long end, ParseOptions const& opts,
     start++;
   }
 
+  // Skip over the "0x" prefix for hex notation
+  if (base == 16 && start + 2 <= end && data[start] == '0' && data[start + 1] == 'x') {
```

Comment:
Maybe it would be a good to add an optional prefix string to ParseOptions, so you could also potentially support things like `#` in hex color strings (`#7400ff`), `0b` in binary strings, etc

Reply:
Perhaps, but I'm definitely not doing this as part of this PR :) ~ really, the profile results show that the parsing code needs a complete overhaul, so that would be a good time to add functionality (right now, the primary goal is to bring the libcudf++ reader on par with the legacy reader)

Reply:
Also, does pandas automatically parse hex numbers when the prefix is present or does it require the base/hex to be explicitly specified? It seems like it would be easy to automatically override the base to 16 here if the first two chars are `0x`, and otherwise parse as base 10 or whatever, for example.

Reply:
I would assume pandas infers the base automatically based on the prefix for integer columns, but cudf has explicit `"hex" / "hex32" / "hex64"` dtypes (probably something that should be revisited for improving pandas compatibility)

Reply:
I'm not sure is 0x is required for hex numbers. Users can specify a column to be hex, so maybe in that case the prefix is not needed.